### PR TITLE
[Configs] feat: Add staking config parser of gateway staking

### PR DESCRIPTION
**Pull request by red-0ne**
### Human Summary

* Add a staking config option to the `poktrolld tx stake-gateway` command.
* Implement the gateway staking config parser and unit test it
* Adapt `tx_stake_gateway_test.go` to use a config file instead of the cmd argument

### Summary generated by Reviewpad on 09 Jan 24 17:07 UTC
This pull request introduces several changes across multiple files. Here is a summary of the changes:

1. The file `gateway3_stake_config.yaml` has been added. It contains a single change which is the addition of the `stake_amount` key with a value of `1000upokt`.

2. The `Makefile` has been modified. The `gateway_stake` target now includes a `--config` flag to specify a config file for the stake. The previous targets `gateway1_stake`, `gateway2_stake`, and `gateway3_stake` now specify different config files corresponding to each gateway. Additionally, the `gateway_unstake` target is mentioned, but there is no associated diff for it.

3. The file `tx_stake_gateway_test.go` has undergone several changes. These include the addition of import statements, the addition of a new field `inputConfig` and its corresponding test cases, the writing of `inputConfig` to a temporary file, and modifications to CLI command arguments, error handling, and assertions based on the changes in the test cases.

4. A new file called `gateway_config_reader_test.go` has been added. This file contains a test function `Test_ParseGatewayStakeConfig` that tests the parsing of a gateway stake configuration. The test function includes multiple test cases covering various scenarios and uses the `require` package for assertions and the `yaml` package for normalizing YAML indentation.

5. Another new file named `gateway_config_reader.go` has been added in the `x/gateway/client/config` directory. This file includes package imports, declares data structures, and implements the `ParseGatewayConfig` function for parsing gateway stake configurations. The function validates the stake amount and the denom before returning a `GatewayStakeConfig` object or an error.

6. The file `gateway1_stake_config.yaml` has been added. It contains a single line specifying a stake amount of `1000upokt`.

7. The file `gateway2_stake_config.yaml` has been added. It contains a single line of code that sets the `stake_amount` variable to `1000upokt`.

8. The file `e2e/tests/init_test.go` has the addition of an import statement and a new function `TheUserStakesAWithUpoktFromTheAccount` which deals with creating and using temporary config files for stake operations.

9. The file `cmd/cmd_stake_gateway.go` has been modified with changes including import statements and modifications to the `CmdStakeGateway` function.

10. Lastly, a new file called `errors.go` has been added in the `x/gateway/client/config` directory. This file includes package imports and the registration of several errors related to the gateway client configuration.

Please review these changes and provide feedback as necessary.

## Issue

- #224
- #293 

## Type of change

Select one or more:

- [ ] New feature, functionality or library
- [ ] Bug fix
- [x] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

- [x] **Run all unit tests**: `make go_develop_and_test`
- [x] **Run E2E tests locally**: `make test_e2e`
- [ ] **Run E2E tests on DevNet**: Add the `devnet-test-e2e` label to the PR. This is VERY expensive, only do it after all the reviews are complete.

## Sanity Checklist

- [x] I have tested my changes using the available tooling
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, updated documentation and left TODOs throughout the codebase

### CODE REVIEW

File: x/gateway/client/cli/tx_stake_gateway.go [hunk 1]
```diff
@@ -1,41 +1,51 @@
...
+		Use:   "stake-gateway --config <config_file.yaml>",
 		Short: "Stake a gateway",
 		Long: `Stake a gateway with the provided parameters. This is a broadcast operation that
 will stake the tokens and associate them with the gateway specified by the 'from' address.
 Example:
-$ poktrolld --home=$(POKTROLLD_HOME) tx gateway stake-gateway 1000upokt --keyring-backend test --from $(GATEWAY) --node $(POCKET_NODE)`,
-		Args: cobra.ExactArgs(1),
-		RunE: func(cmd *cobra.Command, args []string) (err error) {
-			clientCtx, err := client.GetClientTxContext(cmd)
+$ poktrolld --home=$(POKTROLLD_HOME) tx gateway stake-gateway --config stake_config --keyring-backend test --from $(GATEWAY) --node $(POCKET_NODE)`,
```
Comment (Olshansk) [inline x/gateway/client/cli/tx_stake_gateway.go:28]:
```suggestion
$ poktrolld --home=$(POKTROLLD_HOME) tx gateway stake-gateway --config stake_config.yaml --keyring-backend test --from $(GATEWAY) --node $(POCKET_NODE)`,
```

**Review by Olshansk — COMMENTED** [review]
1 tiny comment

**Review by Olshansk — APPROVED** [review]

**Comment by red-0ne** [comment]
Finally opted for creating the `gateway_stake_config.yaml` file from withing the e2e step, I think it's the most appropriate way to do it as the test is having the stake amount as input anyway.

**Comment by red-0ne** [comment]
> > Finally opted for creating the `gateway_stake_config.yaml` file from withing the e2e step, I think it's the most appropriate way to do it as the test is having the stake amount as input anyway.
> 
> That works! I would prefer us to also have an option to feed the config via `stdin`, but creating a file works as well. Just need to make sure the file is in `.gitignore` to avoid committing it accidentally.

It's a temp. file that is deleted as soon as the test suite finishes.

But totally agree about the `stdin` feature :+1:

**Comment by Olshansk** [comment]
@red-0ne

>  But totally agree about the stdin feature 👍

How about a `community` ticket? To make it easier, you can screenshot & link to this PR for context.

File: e2e/tests/init_test.go [hunk 2]
```diff
@@ -166,18 +167,36 @@ func (s *suite) TheUserShouldWaitForSeconds(dur int64) {
 }
 
 func (s *suite) TheUserStakesAWithUpoktFromTheAccount(actorType string, amount int64, accName string) {
```
Comment (Olshansk) [inline e2e/tests/init_test.go:169]: I LOVE THAT OUR E2E TESTS CAUGHT THIS!

**Pull request merged**